### PR TITLE
Adding Zoom & Slack fake client installer HTML detections

### PR DESCRIPTION
**Pull request by aidenmitchell**

### CODE REVIEW

**Review by morriscode — COMMENTED** [review]
I really like the intent 👍  I couldn't find anything concrete on if these are being delivered as attachments or not. It's completely plausible to embed these in the HTML email bodies in which case these rules miss them.  If the intent is to find HTML attachments we can restrict to the attachment type? If it's both, then it's too narrow. Does that make sense? 

I took a look at the logic on the HTML binexplode and I think it could be done with positional inspections over strings. Specifically both samples have static titles containing slack or zoom and url paths with the exe in them. The below matches both delivr.to samples as HTML attachments... and in the HTML body.  If you wanted to play around with it. I'm not opposed to combining the logic and adding brands for Fake Downloaders, or keeping them split like you had them by brand. I also tried to see if we had any brand confidence on these but came up empty! Happy to chat


Playground link to HTML in body : https://playground.sublimesecurity.com?id=cfbb3b71-b30b-42c1-bf4b-87e55536b65d

```
  type.inbound 
  and sender.email.domain.root_domain not in ("slack.com", "zoom.us")

  and (
    body.html.display_text != ""
    and strings.ilike(body.html.display_text, 
      "*Slack*for*Windows*", "*zoom*")
    and strings.ilike(body.html.display_text, "*download*")
  
  or any(attachments,
        (
            .file_extension in~ ("html", "htm", "shtml", "dhtml") or
            .file_extension in~ $file_extensions_common_archives or
            .file_type == "html"
        )
        and any(beta.binexplode(.),
                strings.ilike(.scan.html.title, "*zoom*", "*slack*") and
                any(.scan.url.urls, strings.ilike(.path, '*exe*', '*msi*', '*zip*'))
              )
  
  )
)
```

**Comment by aidenmitchell** [comment]
> If the intent is to find HTML attachments we can restrict to the attachment type? If it's both, then it's too narrow.

Detecting both the attachments and inline HTML in one rule is probably the better route. I was too focused on the delivr.to attachment method 😛 
Good catch, thanks!

> keeping them split like you had them by brand

I think we should keep them split-up; in the event that one brand's detection gets a little complicated, keeping them separate makes it easy on us. I also think that from a user perspective, it'll be nice to see what brand was specifically flagged.


I'll take what you have in that Playground link and split it into 2 rules if that's cool!

**Comment by morriscode** [comment]
🔥  Thanks @aidenmitchell  Let me grab these and test them. I'm skeptical on the body detection standing on it's own (I know i'm the one who put it there haha) open to ideas if you have any. I feel like these could be prone to FP's from discord/community service mailers. I'll run them and see how they go!

**Comment by morriscode** [comment]
Testing confirmed, the body match is too loose. Several hundred FP's on threaded messages with users discussing Zoom/Slack, newsletters and bulk mailers.

**Comment by aidenmitchell** [comment]
I'll look into this one, it also looks like the Syntax for the Zoom one is wrong, it's not flagging my sample 😛

File: detection-rules/malicious_slack_html.yml [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    any(beta.binexplode(beta.linkanalysis(.).screenshot),
+        all([
+            "download",
+            "app",
+            "version"
+        ], strings.icontains(..scan.ocr.raw, .))
+    )
+    )
+   )
+tags:
```
Comment (morriscode) [inline detection-rules/malicious_slack_html.yml:41]:
```suggestion
name: "Malicious Slack download attachment"
description: |
  Impersonation of the video conferencing provider Slack, encouraging the user to download a fake Slack client.
type: "rule"
references:
  - "https://delivr.to/payloads?id=790c79d0-3ecc-4bcf-aa51-f417c65c5983"
severity: "high"
source: |
  type.inbound 
  and sender.email.domain.root_domain not in ("slack.com")
  and (
    (
      body.html.display_text != ""
      and strings.ilike(body.html.display_text, "*Slack*for*Windows*")
      and strings.ilike(body.html.display_text, "*download*")
    )
    or (
      any(attachments,
        (
          .file_extension in~ ("html", "htm", "shtml", "dhtml") or
          .file_extension in~ $file_extensions_common_archives or
          .file_type == "html"
        )
        and any(beta.binexplode(.),
          strings.ilike(.scan.html.title, "*slack*") and
          any(.scan.url.urls, strings.ilike(.path, '*exe*', '*msi*', '*zip*'))
        )
      )
    )
    and (
      any(body.links,
        beta.linkanalysis(.).credphish.brand.name == "Slack" and 
        beta.linkanalysis(.).credphish.brand.confidence == "high" and 
        beta.linkanalysis(.).credphish.disposition == "phishing" and 
        any(beta.binexplode(beta.linkanalysis(.).screenshot),
          all([
            "download",
            "app",
            "version"
          ], strings.icontains(..scan.ocr.raw, .))
        )
      )
    )
  )
tags:
  - "Brand impersonation"
```
@aidenmitchell  What do you think about moving that second OR to an AND. This way we can strengthen the body text anchor to have an AND on the body.links section being true also?  I think this logically makes sense, please sanity check me :)

Comment (morriscode) [reply detection-rules/malicious_slack_html.yml:41]: This isn't logically operating the way I want it to in testing. i'll ping you in slack about my thoughts for next steps.

**Comment by aidenmitchell** [comment]
Closing, we're going to refactor this one.

**Pull request closed without merging**